File: CoreAPI/UIModels/Feature.cs
using System;
using System.Collections.Generic;

namespace CoreAPI.UIModels;

public partial class Feature
{
    public string Id { get; set; }

    public string Name { get; set; }

    public string Label { get; set; }

    public string ParentId { get; set; }

    public int? Order { get; set; }

    public string ClassName { get; set; }

    public string Style { get; set; }

    public string StyleSheet { get; set; }

    public string Script { get; set; }

    public string Events { get; set; }

    public string Icon { get; set; }

    public bool IsDevider { get; set; }

    public bool IsGroup { get; set; }

    public bool IsMenu { get; set; }

    public bool IsPublic { get; set; }

    public bool StartUp { get; set; }

    public string ViewClass { get; set; }

    public string EntityId { get; set; }

    public string Description { get; set; }

    public bool? Active { get; set; }

    public DateTime? InsertedDate { get; set; }

    public string InsertedBy { get; set; }

    public DateTime? UpdatedDate { get; set; }

    public string UpdatedBy { get; set; }

    public bool IsSystem { get; set; }

    public bool IgnoreEncode { get; set; }

    public bool InheritParentFeature { get; set; }

    public bool DeleteTemp { get; set; }

    public bool CustomNextCell { get; set; }

    public bool LoadEntity { get; set; }

    public bool IsLock { get; set; }

    public bool IsFlow { get; set; }

    public string CodeId { get; set; }

    public string AppId { get; set; }
}

File: frontend/lib/models/feature.js
import { Component } from "./component";
import { UserSetting } from "./userSetting";

/**
 * Represents a feature.
 * @class
 */
export class Feature {
    /** @type {string} */
    id = '';
    /** @type {string} */
    name = '';
    /** @type {string} */
    label = '';
    /** @type {string} */
    parentId = '';
    /** @type {number} */
    order = 0;
    /** @type {string} */
    className = '';
    /** @type {string} */
    style = '';
    /** @type {string} */
    styleSheet = '';
    /** @type {string} */
    script = '';
    /** @type {string} */
    events = '';
    /** @type {string} */
    icon = '';
    /** @type {boolean} */
    isDivider = false;
    /** @type {boolean} */
    isGroup = false;
    /** @type {boolean} */
    isMenu = false;
    /** @type {boolean} */
    isPublic = false;
    /** @type {boolean} */
    startUp = false;
    /** @type {string} */
    viewClass = '';
    /** @type {string} */
    entityId = '';
    /** @type {string} */
    description = '';
    /** @type {boolean} */
    active = false;
    /** @type {Date} */
    insertedDate = new Date();
    /** @type {string} */
    insertedBy = '';
    /** @type {Date} */
    updatedDate = new Date();
    /** @type {string} */
    updatedBy = '';
    /** @type {boolean} */
    isSystem = false;
    /** @type {boolean} */
    ignoreEncode = false;
    /** @type {boolean} */
    inheritParentFeature = false;
    /** @type {boolean} */
    deleteTemp = false;
    /** @type {boolean} */
    customNextCell = false;
    /** @type {boolean} */
    isFullScreen = false;
    /** @type {boolean} */
    isSmallScreen = false;
    /** @type {boolean} */
    loadEntity = false;
    /** @type {boolean} */
    isLock = false;
    /** @type {string} */
    html = '';
    /** @type {Component[]} */
    components = [];
    /** @type {any} */
    layout;
    /** @type {HTMLElement} */
    parentElement;
    /** @type {Feature[]} */
    inverseParent = [];
    /** @type {UserSetting[]} */
    userSettings = [];
}